salt/states/zabbix_host.py

```diff
@@ -336,21 +351,33 @@ def _interface_format(interfaces_data):
         elif not hostinterfaces and interfaces:
             update_interfaces = True
 
-        cur_inventory = __salt__["zabbix.host_inventory_get"](
-            hostids=hostid, **connection_args
-        )
-        if cur_inventory:
-            # Remove blank inventory items
-            cur_inventory = {k: v for k, v in cur_inventory.items() if v}
-            # Remove persistent inventory keys for comparison
-            cur_inventory.pop("hostid", None)
-            cur_inventory.pop("inventory_mode", None)
-
-        if not cur_inventory:
-            if new_inventory:
-                update_inventory = True
-        elif set(cur_inventory) != set(new_inventory):
-            update_inventory = True
+        # if inventory param is empty leave inventory as is don't compare it
+        # if inventory_mode is '-1', the inventory will be erased, why compare it?
+        if inventory is not None and inventory_mode != "-1":
```

Comment:
**question** aren't we making `inventory_mode` something else? Or is that only when the *host* provides -1?

Reply:
Yes... I guess this is becoming more confused as time is passing.

This is the `inventory_mode` from state declaration. The override of `-1` is only if the `host` provides it. This override is just to pass this test and don't compare the inventory if it's set to be erased (`-1` in state declaration) but compare inventory if it's currently disabled (`-1` in host) and being populated now.

Reply:
Did the change in `inventory_mode` handling. Hope it's now more comprehensible.

Reply:
:+1: This seems to be reasonable